src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java

```diff
@@ -128,42 +137,67 @@ private Boolean resolveGDPRApplies(Regs regs) {
         return gdpr != null ? gdpr == 1 : Boolean.TRUE;
     }
 
-    private InvibesInternalParams updateInvibesInternalParams(
-            InvibesInternalParams invibesInternalParams, ExtImpInvibes invibesExt,
-            Imp imp) {
-        final String placementId = invibesExt.getPlacementId();
-        final List<String> updatedPlacementIds = getUpdatedPlacementIds(
-                placementId, invibesInternalParams.getBidParams().getPlacementIDs());
-
-        final Map<String, InvibesPlacementProperty> updatedProperties
-                = getUpdatedProperties(invibesInternalParams.getBidParams().getProperties(),
-                placementId, imp.getId(), resolveAdFormats(imp.getBanner()));
+    private InvibesInternalParams updateInvibesInternalParams(InvibesInternalParams invibesInternalParams,
+                                                              ExtImpInvibes invibesExt,
+                                                              Imp imp) {
+        final String impExtPlacementId = invibesExt.getPlacementId();
+        final InvibesBidParams bidParams = invibesInternalParams.getBidParams();
+        final List<String> updatedPlacementIds = bidParams.getPlacementIds();
 
-        final InvibesBidParams updatedBidParams = invibesInternalParams.getBidParams().toBuilder()
-                .placementIDs(updatedPlacementIds)
-                .properties(updatedProperties)
+        if (StringUtils.isNotBlank(impExtPlacementId)) {
+            updatedPlacementIds.add(impExtPlacementId.trim());
+        }
+        final Banner banner = imp.getBanner();
+        final List<Format> adFormats = resolveAdFormats(banner);
+
+        bidParams.getProperties()
+                .put(impExtPlacementId, InvibesPlacementProperty.builder()
+                        .impId(imp.getId())
+                        .formats(adFormats)
+                        .build());
+
+        final InvibesBidParams updatedBidParams = bidParams.toBuilder()
+                .placementIds(updatedPlacementIds)
+                .properties(bidParams.getProperties())
```

Comment:
this line is redundant if you use bidParams.toBuilder(). 
Are you sure that you want to use builder here. You can set return value to void, and modify received object